File: app/src/main/java/com/doapps/habits/models/IProgramViewProvider.java
package com.doapps.habits.models;

import com.google.firebase.database.DataSnapshot;

public interface IProgramViewProvider {
    /**
     * Returns title of Program
     * @return Title of {@link com.doapps.habits.models.Program}
     *
     */
    String getTitle();
    /**
     * @return Description of {@link com.doapps.habits.models.Program}
     */
    String getDescription();
    /**
     * @return Success rate of {@link com.doapps.habits.models.Program}
     */
    String getPercent();
    /**
     * @return Link to image of {@link com.doapps.habits.models.Program}
     */
    String getImageLink();
    /**
     * @return Data snapshot of {@link com.doapps.habits.models.Program}
     */
    DataSnapshot getSnapshot();
}

File: app/src/main/java/com/doapps/habits/models/IProgramViewProvider.kt
package com.doapps.habits.models

import com.google.firebase.database.DataSnapshot

/**
 * An interface for providers of program view
 */
interface IProgramViewProvider {

  /**
   * Returns title of Program
   *
   * @return Title of [Program]
   */
  val title: String?

  /**
   * @return Description of [Program]
   */
  val description: String?

  /**
   * @return Success rate of [Program]
   */
  val percent: String?

  /**
   * @return Link to image of [Program]
   */
  val imageLink: String?

  /**
   * @return Data snapshot of [Program]
   */
  val snapshot: DataSnapshot
}
